### PR TITLE
[SPARK-46611][CORE] Remove ThreadLocal by replace SimpleDateFormat with DateTimeFormatter

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
This PR propose to remove `ThreadLocal` by replace `SimpleDateFormat` with `DateTimeFormatter`.

### Why are the changes needed?
`SimpleDateFormat` is not thread safe, so we wrap it with `ThreadLocal`.
`DateTimeFormatter` is thread safe, we can use it instead.
According to the javadoc of `SimpleDateFormat`, it recommended to use `DateTimeFormatter` too.

In addition, `DateTimeFormatter` have better performance than `SimpleDateFormat` too.

### Does this PR introduce _any_ user-facing change?
'No'.

### How was this patch tested?
GA tests.

### Was this patch authored or co-authored using generative AI tooling?
'No'.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/ui/UIUtils.scala [hunk 1]
```diff
@@ -42,15 +44,18 @@ private[spark] object UIUtils extends Logging {
   val TABLE_CLASS_STRIPED = TABLE_CLASS_NOT_STRIPED + " table-striped"
   val TABLE_CLASS_STRIPED_SORTABLE = TABLE_CLASS_STRIPED + " sortable"
 
-  // SimpleDateFormat is not thread-safe. Don't expose it to avoid improper use.
-  private val dateFormat = new ThreadLocal[SimpleDateFormat]() {
-    override def initialValue(): SimpleDateFormat =
-      new SimpleDateFormat("yyyy/MM/dd HH:mm:ss", Locale.US)
-  }
+  private val dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy/MM/dd HH:mm:ss", Locale.US)
```
Comment (LuciferYang) [inline core/src/main/scala/org/apache/spark/ui/UIUtils.scala:47]: How about 

```scala
private val dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy/MM/dd HH:mm:ss", Locale.US)
    .withZone(java.time.ZoneId.systemDefault())

def formatDate(date: Date): String = dateTimeFormatter.format(date.toInstant)
def formatDate(timestamp: Long): String = dateTimeFormatter.format(Instant.ofEpochMilli(timestamp))
```

File: core/src/main/scala/org/apache/spark/ui/UIUtils.scala [hunk 2]
```diff
@@ -42,15 +44,18 @@ private[spark] object UIUtils extends Logging {
...
+    val localDateTime = date.toInstant.atZone(java.time.ZoneId.systemDefault()).toLocalDateTime
+    localDateTime.format(dateTimeFormatter)
+  }
 
-  def formatDate(timestamp: Long): String = dateFormat.get.format(new Date(timestamp))
+  def formatDate(timestamp: Long): String = {
+    val localDateTime =
+      Instant.ofEpochMilli(timestamp).atZone(java.time.ZoneId.systemDefault()).toLocalDateTime
+    localDateTime.format(dateTimeFormatter)
+  }
```
Comment (mridulm) [inline core/src/main/scala/org/apache/spark/ui/UIUtils.scala:58]: nit: cache `java.time.ZoneId.systemDefault()` as a field.

**Review by mridulm — COMMENTED** [review]
This is a nice change.
There are other usages of `SimpleDateFormat` which are candidates for this pattern.

File: core/src/main/scala/org/apache/spark/ui/UIUtils.scala (hunk 1 above)
Comment (mridulm) [reply core/src/main/scala/org/apache/spark/ui/UIUtils.scala:47]: This is even better than my suggestion, thanks @LuciferYang !

File: core/src/main/scala/org/apache/spark/ui/UIUtils.scala (hunk 2 above)
Comment (mridulm) [reply core/src/main/scala/org/apache/spark/ui/UIUtils.scala:58]: @LuciferYang's suggestion [here](https://github.com/apache/spark/pull/44613#discussion_r1443097993) is better.

**Comment by beliefer** [comment]
> There are other usages of `SimpleDateFormat` which are candidates for this pattern.

I will check.

**Comment by beliefer** [comment]
ping @dongjoon-hyun @srowen

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195431809。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195431809.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195432886。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195432886.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195434035。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195434035.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195434855。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195434855.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195436316。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195436316.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195437543。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195437543.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195438601。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195438601.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195439902。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195439902.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195441010。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195441010.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195442305。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195442305.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195443373。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195443373.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195444707。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195444707.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195445869。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195445869.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195447327。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195447327.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195448316。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195448316.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195449442。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195449442.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195451089。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195451089.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195452405。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195452405.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195453736。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195453736.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Review by srowen — APPROVED** [review]

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195454870。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195454870.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195455952。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195455952.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195456973。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195456973.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195458623。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195458623.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195459708。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195459708.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195460870。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195460870.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195462227。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195462227.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195463590。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195463590.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195464943。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195464943.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195465744。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195465744.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195466919。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195466919.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195468388。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195468388.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by MaxGekk** [comment]
I sent info about the spam from @llkj1 to GitHub support.

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195528798。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195528798.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195530026。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195530026.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by srowen** [comment]
I tried to block the user and have been deleting the spam

**Comment by LuciferYang** [comment]
is this one ready to go?

**Review by LuciferYang — APPROVED** [review]
+1, LGTM

**Comment by LuciferYang** [comment]
Merged into master. Thanks @beliefer@srowen @mridulm

File: core/src/main/scala/org/apache/spark/ui/UIUtils.scala [hunk 3]
```diff
@@ -150,19 +147,19 @@ private[spark] object UIUtils extends Logging {
...
       showYYYYMMSS: Boolean = true,
       timezone: TimeZone = null): String = {
-    val oldTimezones =
-      (batchTimeFormat.get.getTimeZone, batchTimeFormatWithMilliseconds.get.getTimeZone)
+    val oldTimezones = (batchTimeFormat.getZone, batchTimeFormatWithMilliseconds.getZone)
     if (timezone != null) {
-      batchTimeFormat.get.setTimeZone(timezone)
-      batchTimeFormatWithMilliseconds.get.setTimeZone(timezone)
+      val zoneId = timezone.toZoneId
+      batchTimeFormat.withZone(zoneId)
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/ui/UIUtils.scala:153]: This looks like a regression - see https://github.com/apache/spark/pull/44619

**Comment by beliefer** [comment]
@LuciferYang @srowen @mridulm Thank you!

**Pull request closed without merging**